packages/reactivity/__tests__/computed.spec.ts

```diff
@@ -322,5 +322,42 @@ describe('reactivity/computed', () => {
       // triggering the final subscribing effect.
       expect(c2Spy).toHaveBeenCalledTimes(4)
     })
+
+    test('chained computed value on-demand trigger', async () => {
+      const c1Spy = jest.fn()
+      const c2Spy = jest.fn()
+
+      const src = ref(0)
+      const c1 = computed(() => {
+        c1Spy()
+        return src.value < 5
+      })
+      const c2 = computed(() => {
+        c2Spy()
+        return c1.value ? '< 5' : '>= 5'
+      })
+
+      expect(c1Spy).toHaveBeenCalledTimes(0)
+      expect(c2Spy).toHaveBeenCalledTimes(0)
+
+      expect(src.value).toBe(0)
+      expect(c2.value).toBe('< 5')
+      expect(c1Spy).toHaveBeenCalledTimes(1)
+      expect(c2Spy).toHaveBeenCalledTimes(1)
+
+      for (let i = 0; i < 10; i++) {
+        src.value++
+      }
+      expect(src.value).toBe(10)
+      expect(c2.value).toBe('>= 5')
+      expect(c1Spy).toHaveBeenCalledTimes(2)
+      expect(c2Spy).toHaveBeenCalledTimes(2)
+
+      src.value++
+      expect(src.value).toBe(11)
+      expect(c2.value).toBe('>= 5')
+      expect(c1Spy).toHaveBeenCalledTimes(3)
+      expect(c2Spy).toHaveBeenCalledTimes(2) // failed here
```

Comment:
After 9086bb40 most cases would work as expected - this would still fail because when `c2.value` was synchronously accessed on line 358, `c1` isn't even calculated yet so we need to force computed `c2` which in turn force computes `c1`.

If you remove the sync `c2.value` access on line 358 then `c2` won't have to re-computed. I think this is an ok trade-off.